### PR TITLE
Major versions upgrade

### PR DESCRIPTION
**Pull request by fredjn**
### Applicable Issues
fixes: #65
fixes: #64
fixes: #40

### Description of the Change
- Major Python and dependency version(s) overhaul. Moving away from python 3.6 has been long overdue, and these changes bump us all the way to 3.11, which is the latest and greatest.
- Add docker compose file to help ease the setup of a working dev environment, which include
    - RabbitMQ
    - MongoDB
    - GraphQL API
    - Storage backend

### Alternate Designs
Moving everything to Poetry might have been even better, but I figured it should be handled in a separate PR.

### Benefits
Running EOL versions of python and friends is bad, mkay!

### Possible Drawbacks
Since it is quite a big version bump there might be runtime and intermittent bugs ahead, but I don't see how it can warrant running EOL versions of software.

### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by: Fredrik Fristedt <<fredrik.fristedt@axis.com>>

### CODE REVIEW

File: deploy/development.env [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+RABBITMQ_USERNAME=guest
+RABBITMQ_PASSWORD=guest
+RABBITMQ_PORT=5672
+RABBITMQ_VHOST=
+RABBITMQ_QUEUE=graphql-test
+RABBITMQ_DURABLE_QUEUE=true
+MONGODB_CONNSTRING=mongodb://root:pass@mongodb/admin
+MONGODB_DATABASE=admin
+MONGODB_INITDB_ROOT_USERNAME=root
+MONGODB_INITDB_ROOT_PASSWORD=pass
```
Comment (magnusbaeck) [inline deploy/development.env:12]: Missing newline at EOF.

File: deploy/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    build:
+      context: ../
+      dockerfile: Dockerfile.storage
+    networks:
+      - eiffel-graphql-api-network
+    env_file:
+      - development.env
+    depends_on:
+      - rabbitmq
+      - mongodb
```
Comment (magnusbaeck) [inline deploy/docker-compose.yml:44]: Missing newline at EOF.

File: src/eiffel_graphql_api/graphql/schemas/links/artifact.py [hunk 3]
```diff
@@ -43,9 +43,45 @@ def resolve_artifact_created(self, _):
...
+    def resolve_artifact_created(self, _):
+        """Resolve artifact created link."""
+        from ..union import NotFound  # pylint:disable=import-outside-toplevel
+
+        event = find_one(
+            "EiffelArtifactCreatedEvent", {"meta.id": self.link.get("target")}
+        )
+        if event is None:
+            return NotFound(self.link, "Could not find event in database.")
+        return ArtifactCreated(event)
```
Comment (magnusbaeck) [inline src/eiffel_graphql_api/graphql/schemas/links/artifact.py:87]: Is this related to the package upgrade or something you did on the fly?

**Review by magnusbaeck — COMMENTED** [review]
To get the PR connected to all three issues in the PR description, you need one "Fixes " prefix per issue, i.e.

```
Fixes #1
Fixes #2
Fixes #3
```

File: src/eiffel_graphql_api/graphql/schemas/links/artifact.py (hunk 3 above)
Comment (fredjn) [reply src/eiffel_graphql_api/graphql/schemas/links/artifact.py:87]: It's related to the upgrade, for some reason having the ArtifactPreviousVersion and ReusedArtifact inherit from Artifact stopped working. I could perhaps have spent some time trying to figure out exactly why, but the pain of having a small boiler plate was way less discomforting than digging into the internals of the GraphQL libraries.

File: deploy/docker-compose.yml (hunk 2 above)
Comment (fredjn) [reply deploy/docker-compose.yml]: fixed

File: deploy/development.env (hunk 1 above)
Comment (fredjn) [reply deploy/development.env]: fixed

**Review by magnusbaeck — APPROVED** [review]

File: src/eiffel_graphql_api/graphql/schemas/links/artifact.py (hunk 3 above)
Comment (fredjn) [reply src/eiffel_graphql_api/graphql/schemas/links/artifact.py:87]: No further luck, leaving it as it is for now.

**Pull request merged**